tools/stonith_admin.c

```diff
@@ -387,6 +398,32 @@ show_history(stonith_t *st, const char *target, int timeout, int quiet,
     return rc;
 }
 
+static int
+validate(stonith_t *st, const char *agent, const char *id,
+         stonith_key_value_t *params, int timeout, int quiet)
+{
+    int rc = 1;
+    char *output = NULL;
+    char *error_output = NULL;
+
+    // @TODO allow rsc_id for CIB secrets
+    rc = st->cmds->validate(st, st_opt_sync_call, id, NULL, agent, params,
+                            timeout, &output, &error_output);
+
+    if (!quiet) {
+        printf("Validation of %s %s\n", agent, (rc? "failed" : "succeeded"));
```

Comment:
It does return error for lha fence agents:
```
"Validation of external/sbd failed"
```
-- It would be better if we could probably add the reason of the error in here. Since only running with -VV we could also see the warning:
```
(stonith__lha_validate)         warning: Validation of Linux-HA fence agents is not supported
```

Reply:
Ah, I overlooked that. I worked around a similar case (agent not found) by using crm_perror(), which both logs and prints to stderr, so I'll do the same here.

Reply:
Thanks. But still it only says:
```
Validation of external/libvirt failed
```
;-) And only with -VV, it says:
```
Cannot validate Linux-HA fence agents: Operation not supported (95)
(stonith__lha_validate)         warning: Cannot validate Linux-HA fence agents: Operation not supported (95)                                                                                                     
Validation of external/libvirt failed
```

Reply:
Blah, it probably has to be error to show up without a flag.

Reply:
Updated commit

Reply:
It does the trick now ;) Thanks.